src/vs/workbench/browser/layout.ts

```diff
@@ -1484,6 +1494,17 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 			this.hostService.toggleFullScreen(mainWindow);
 		}
 
+		// restore focus if part is still visible, otherwise fallback to editor
```

Comment:
I think it would be safer to store the window and the focused part together, because maybe they become different here?